WordPress/gutenberg
# Navigation: preload more API requests

## Description
When loading the Navigation screen, there are a lot of loading spinners due to API requests to /menus and /menu-items. We should preload these requests so that the screen appears faster.
The goal of this PR is to preload these requests.
Also, this PR fixes 2 small bugs in the `core-data` and `api-fetch` modules.
Fixes https://github.com/WordPress/gutenberg/issues/24642

## How has this been tested?
1. Open the navigation page (`wp-admin/admin.php?page=gutenberg-navigation`).
2. Make sure there is only 1 fetch/XHR request (see the screenshot below).

## Screenshots 
Before:
![Screenshot 2021-10-06 at 16 44 50](https://user-images.githubusercontent.com/43744263/136235248-16a0f1ae-0324-4e45-98dd-b7b0d155b288.png)
After:
![Screenshot 2021-10-06 at 17 29 11](https://user-images.githubusercontent.com/43744263/136235260-b53acf9c-a0cd-4070-8881-ddeee5c17945.png)

## Types of changes
New feature





## Checklist:
